lms/djangoapps/course_goals/management/commands/goal_reminder_email.py

```diff
@@ -0,0 +1,100 @@
+"""
+Command to trigger sending reminder emails for learners to achieve their Course Goals
+"""
+from datetime import date, timedelta
+
+from django.contrib.sites.models import Site
+from django.core.management.base import BaseCommand
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.certificates.api import get_certificate_for_user_id
+from lms.djangoapps.certificates.data import CertificateStatuses
+from lms.djangoapps.course_goals.models import CourseGoal, UserActivity
+from lms.djangoapps.course_goals.toggles import COURSE_GOALS_NUMBER_OF_DAYS_GOALS
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.lib.celery.task_utils import emulate_http_request
+from openedx.features.course_duration_limits.access import get_user_course_expiration_date
+
+MONDAY_WEEKDAY = 0
+SUNDAY_WEEKDAY = 6
+
+
+def ace_send():
+    """Just here as a mock hook for tests - drop this once we fix AA-909"""
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms goal_reminder_email
+    """
+    help = 'Send emails to users that are falling behind their course goals for the week'
+
+    def handle(self, *args, **options):
+        """
+        Docstring
+
+        Helpful notes for the function:
+            weekday() returns an int 0-6 with Monday being 0 and Sunday being 6
+        """
+        today = date.today()
+        sunday_date = today + timedelta(days=SUNDAY_WEEKDAY - today.weekday())
+        monday_date = today - timedelta(days=today.weekday())
+
+        # Monday is the start of when we consider user's activity towards counting towards their weekly
+        # goal. As such, we use Mondays to clear out the email reminders sent from the previous week.
+        if today.weekday() == MONDAY_WEEKDAY:
+            CourseGoal.objects.filter(email_reminder_sent=True).update(email_reminder_sent=False)
```

Comment:
do we have any concerns about the potential performance of this query when there are a lot of goal records?

Reply:
Good question. The SQL query that django generates here is a one-liner at least:
```
UPDATE `course_goals_coursegoal` SET `email_reminder_sent` = 0 WHERE `course_goals_coursegoal`.`email_reminder_sent` = 1
```

But yeah that could hit a lot of rows. I don't know the performance impact of that vs some other strategy. This is done in the context of a management command, so we don't care how long it takes us, except that we are probably locking the db while this is running.

So other goals being updated might block on the result of this? And yeah... Why have that performance profile at all?

Could have this feature use its own little model - `CourseGoalReminder` or something - and have it just be a 1-to-1 model with a goal and hold the `email_reminder_sent` value there...  What do you think about that?

Reply:
hm, was asking more just because I was curious, but having it in one model seems cleaner if that wouldn't cause issues
we could potentially just ask in the SRE channel if they think this query would be fine but if you think it should be fine that's ok too?

Reply:
I suspect it would be fine (-ish), but I think it might be a better design to split it out anyway. Let this mini-feature have its own model / status tracking that doesn't involve the rest of the feature's data.

Reply:
sure